airbyte-integrations/bases/base-python/base_python/sdk/streams/http.py

```diff
@@ -0,0 +1,252 @@
+import copy
+import requests
+
+from abc import ABC, abstractmethod
+from typing import Mapping, Iterable, Any, Optional
+
+
+from base_python.sdk.streams.auth.core import HttpAuthenticator, NoAuth
+from base_python.sdk.streams.exceptions import DefaultBackoffException, CustomBackoffException
+from base_python.sdk.streams.rate_limiting import custom_backoff_handler, default_backoff_handler
+
+from base_python.sdk.streams.core import Stream
+
+
+class HttpStream(Stream, ABC):
```

Comment:
You have a composition issue here don't you? 

What if you want it to support Incremental?

Reply:
for context: the current way this is doable is through multiple inheritance i.e: extend `IncrementalStream` and `HttpStream`

Reply:
@sherifnada If we assume that every stream needs to handle state to optimize writing at least, we can safely assume that all streams provide same interface and there is no such thing as `IncrementaStream`.

Reply:
merged `IncrementalStream` with `Stream`